File: frontendexpo/tree.js
const fs = require('fs');
const path = require('path');

const ignore = ['node_modules', '.git'];

function tree(dir, prefix = '') {
  const files = fs.readdirSync(dir);
  const filteredFiles = files.filter((file) => !ignore.includes(file));

  filteredFiles.forEach((file, index) => {
    const fullPath = path.join(dir, file);
    const stats = fs.statSync(fullPath);
    const isLast = index === filteredFiles.length - 1;

    console.log(`${prefix}${isLast ? '└─' : '├─'} ${file}`);

    if (stats.isDirectory()) {
      tree(fullPath, `${prefix}${isLast ? '  ' : '│ '}`);
    }
  });
}

tree(process.cwd());

File: calorie_counter/tree.py
import os

ignore = ['venv', '__pycache__', '.git', 'migrations', 'node_modules']

def tree(directory, prefix=''):
    files = os.listdir(directory)
    files = [f for f in files if f not in ignore]
    files.sort()

    for index, file in enumerate(files):
        path = os.path.join(directory, file)
        is_last = index == len(files) - 1

        if os.path.isdir(path):
            print(f"{prefix}{'└─ ' if is_last else '├─ '}{file}")
            tree(path, prefix + ('    ' if is_last else '│   '))
        else:
            print(f"{prefix}{'└─ ' if is_last else '├─ '}{file}")

if __name__ == '__main__':
    tree(os.getcwd())
